src/serialization/string_utils.cpp

```diff
@@ -524,9 +524,28 @@ std::string half_signed_value(int val)
 
 static void si_string_impl_stream_write(std::stringstream &ss, double input) {
 	std::streamsize oldprec = ss.precision();
+#ifdef _MSC_VER
+	// For MSVC, default mode misbehaves, so we use fixed instead.
 	ss.precision(1);
 	ss << std::fixed
 	   << input;
+#else
+	// In default mode, precision sets the number of significant figures.
+
+	// 999.5 and above will render as 1000+, however, only numbers above 1000 will use 4 digits
+	// Rounding everything from 100 up (at which point we stop using decimals anyway) avoids this.
+	if (input >= 100)
+		input = round(input);
```

Comment:
Please use `std::round()`. The function [isn't guaranteed](https://stackoverflow.com/a/2588103) to be in the global namespace.